pkgs/by-name/ca/cato/package.nix

```
@@ -0,0 +1,60 @@
+{
+  stdenv,
+  fetchurl,
+  dpkg,
+  lib,
+  libz,
+  autoPatchelfHook,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cato";
+  version = "5.2.1.1";
+
+  src = fetchurl {
+    url = "https://clients.catonetworks.com/linux/${version}/cato-client-install.deb";
+    sha256 = "sha256-0hUchaxaiKJth2ByQMFfjsCLi/4kl+SrNSQ33Y6r3WA=";
+  };
+
+  dontConfigure = true;
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+  ];
+
+  buildInputs = [
+    libz
+    stdenv.cc.cc
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    dpkg -x $src source
+    cd source
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir $out
+
+    mv usr/lib $out/lib
+    mv lib/systemd $out/lib/systemd
+    substituteInPlace "$out/lib/systemd/system/cato-client.service" --replace /usr/sbin/ "$out/bin/"
+
+    mkdir -p $out/bin
+    mv usr/sbin/* $out/bin
+    mv usr/bin/* $out/bin
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Cato";
+    homepage = "https://www.catonetworks.com/";
```

Perhaps the homage URL could be updated to point to the particular product? E.g.

```suggestion
    description = "Cato Client is a lightweight agent that provides secure zero-trust access to resources everywhere";
    homepage = "https://www.catonetworks.com/platform/cato-client/";
```